lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -2729,10 +2756,29 @@ def _add_image_features(self, node_space: schema.NodeSpace) -> None:
             node_space.features.add(
                 features.ArchitectureSettings(arch=sig.architecture)  # type: ignore
             )
+            if (
+                node_space.disk
+                and node_space.disk.os_disk_type
+                and schema.DiskType.Ephemeral in node_space.disk.os_disk_type
+            ):
+                assert isinstance(node_space.disk, features.AzureDiskOptionSettings)
+                node_space.disk.cached_disk_size = search_space.IntRange(
+                    min=self._get_sig_os_disk_size(azure_runbook.shared_gallery)
```

Comment:
You can use a method called `get_os_disk_size` to call the three methods by different disk type. So the check logic doesn't need to be repeated three times here.

Reply:
Done